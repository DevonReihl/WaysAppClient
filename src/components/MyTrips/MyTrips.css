.my-trips-header {
  padding: 15px;
  font-family: 'Archivo Black', Arial, Helvetica, sans-serif;
  text-align: center;
  width: 100%;
  margin: 0.5em 0 0 0;
  color: white;
  font-size: x-large;
  text-shadow: 2px 2px #000;
  background: rgb(247, 134, 243);
  background: linear-gradient(90deg, #48a7f5 0, #40e9e1 100%);
}

.my-trips {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin-top: 3em;
  margin-left: 0;
  padding-left: 2em;
  padding-right: 3em;
  border-top: 5px solid var(--waysBlue);
  border-radius: 20px;
  min-height: 65vh;
}

.my-trip-info-group {
  display: flex;
  flex-direction: column;
  padding: 25px;
  margin: 1em 0 5.5em 0;
  align-items: center;
  justify-content: center;
}

.my-trip-info {
  box-shadow: 0px 1px 0px 0px #ffffff;
  border: 2px solid rgb(46, 140, 216);
  background-color: rgb(46, 140, 216);
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 18px;
  font-weight: 900;
  padding: 4px;
  text-decoration: none;
  width: 155px;
  text-align: center;
}

.modal-content {
  background-color: #fefefe;
  margin-right: 10px;
  padding: 20px;
  text-align: center;
  width: 100%;
}

ul {
  list-style: none;
}

.my-trip-info:hover {
  background-color: white;
  color: rgb(46, 140, 216);
  border: 2px solid rgb(46, 140, 216);
}

.dropdown {
  color: #141414;
  font-family: 'Archivo Black', Arial, Helvetica, sans-serif;
  font-size: 15px;
  font-weight: 300;
  text-align: center;
  align-items: center;
  justify-content: space-around;
  display: none;
  z-index: 1;
  padding: 100px;
  left: 0;
  width: 100%;
  height: auto;
  overflow: auto;
  background-color: rgb(152, 247, 250);
  background-color: rgba(152, 247, 250, 0.4);
}

.my-trip-dropdown:hover .dropdown {
  display: block;
}

.my-trip-cards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: flex-start;
  padding: 1em;
}

.edit-delete-trip-buttons {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.delete-btn {
  width: 7em !important;
}

.delete-btn,
.edit-btn {
  width: 5em;
  font-size: 1em;
  text-align: center;
  margin: 0.4em 0 0 0;
  padding: 0;
}

.my-trip-card-wrapper {
  margin: 1em;
}

@media only screen and (min-width: 600px) {
  .my-trip-info-group {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .dropdown {
    width: 70%;
    left: 10%;
    position: absolute;
  }
}
