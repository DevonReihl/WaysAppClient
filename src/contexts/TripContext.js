import React from 'react';
import UserContext from './UserContext';

const TripContext = React.createContext({
  trips: [],
  currTripId: null,
  setCurrTripId: () => {},
  setTrips: () => {},
  verifyAuth: () => {},
  setLoading: () => {},
  returnUserID: () => {},
  setTripRating: () => {},
  loading: null,
});

export default TripContext;

export class TripProvider extends React.Component {
  state = {
    trips: [],
    currTripId: null,
    currPlace: {},
    stops: [],
    loading: false,
  };

  static contextType = UserContext;

  setTrips = (res) => {
    this.setState({
      trips: res,
    });
  };

  setTripRating = (trip_id) => {
    const index = this.state.trips.findIndex((trip) => {
      return trip.id === Number(trip_id);
    });
    let trips = [...this.state.trips];
    trips[index] = {
      ...trips[index],
      rating: Number(trips[index].rating) + 1,
    };
    this.setState({ trips });
  };

  setLoading = (boolean) => {
    this.setState({
      loading: boolean,
    });
  };

  setCurrTripId = (id) => {
    this.setState({
      currTripId: id,
    });
  };

  handleAddTrip = (trip) => {
    this.setState({
      trips: [...this.state.trips, trip],
    });
  };

  verifyAuth = (id) => {
    return id === this.context.user.id;
  };

  returnUserID = () => {
    return this.context.user.id;
  };

  render() {
    const value = {
      trips: this.state.trips,
      currTripId: this.state.currTripId,
      setTrips: this.setTrips,
      setCurrTripId: this.setCurrTripId,
      verifyAuth: this.verifyAuth,
      loading: this.state.loading,
      setLoading: this.setLoading,
      returnUserID: this.returnUserID,
      setTripRating: this.setTripRating,
    };

    return (
      <TripContext.Provider value={value}>
        {this.props.children}
      </TripContext.Provider>
    );
  }
}
