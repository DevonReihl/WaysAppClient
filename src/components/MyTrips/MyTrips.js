import React, { Component } from 'react';
import TripContext from '../../contexts/TripContext';
import TripCards from '../TripCards/TripCards';
import TokenService from '../../services/token-service';
import { Link } from 'react-router-dom';
import './MyTrips.css';
import TripApiService from '../../services/trip-service';

export default class MyTrips extends Component {
  static contextType = TripContext;

  state = {
    stops: [],
  };

  componentDidMount() {
    let jwt = TokenService.getAuthToken();
    const user = TokenService.parseJwt(jwt);

    TripApiService.getAllStops(user.user_id).then((res) => {
      this.setState({ stops: res });
    });
  }

  renderStates = () => {
    let stops = this.state.stops;
    return stops.map((stop, index) => (
      <li key={index}>
        {stop.city}, {stop.state}
      </li>
    ));
  };

  render() {
    let jwt = TokenService.getAuthToken();
    const user = TokenService.parseJwt(jwt);
    const userTrips = this.context.trips.filter(
      (trip) => trip.user_id === user.user_id
    );

    let count = 0;
    const tripCards = userTrips.map((trip, index) => {
      count++;
      return (
        <div key={trip.id} className="my-trip-card-wrapper">
          <TripCards
            id={trip.id}
            index={index}
            days={trip.days}
            rating={trip.rating}
            destination={trip.destination}
            activities={trip.activities}
            short_description={trip.short_description}
            image={trip.img}
          />
        </div>
      );
    });
    return (
      <>
        <section>
          <div className="my-trips-header">
            <h2>My Trips</h2>
          </div>
        </section>
        <section className="my-trip-info-group">
          <div className="my-trip-dropdown">
            <div className="bttn">Trips Stats</div>
              <div className="dropdown">
                <div className="modal-content">
                  <ul>City, States visited:
                    {this.renderStates()}
                  </ul>
                  <ul>Total trips:
                    <li> {count}</li>
                  </ul>
                </div> 
              </div>
            </div>
          
          <div className="addTripButton">
            <Link to="/add-trip">
              <div className="bttn">Add a Trip!</div>
            </Link>
          </div>
        </section>
        <section className="my-trips">
          <div className="my-trip-cards">{tripCards}</div>
        </section>
      </>
    );
  }
}
