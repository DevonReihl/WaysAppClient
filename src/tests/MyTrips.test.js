import React from 'react'
import MyTrips from '../components/MyTrips/MyTrips.js'
import { shallow } from 'enzyme'
import TripContext from '../contexts/TripContext'
import toJson from 'enzyme-to-json'


describe(`MyTrips component`, () => {
    it('Renders without crashing', () => {
        shallow(
        <TripContext.Provider>
            <MyTrips />
         </TripContext.Provider>
        )
    })
    it('Matches snapshot', () => {
        const wrapper =  shallow(
        <TripContext.Provider>
        <MyTrips />
        </TripContext.Provider>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})