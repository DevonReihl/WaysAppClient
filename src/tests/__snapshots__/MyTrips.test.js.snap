// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTrips component Matches snapshot 1`] = `<MyTrips />`;
